akka-projection-core/src/main/scala/akka/projection/internal/InternalProjectionState.scala

```diff
@@ -298,7 +300,11 @@ private[akka] abstract class InternalProjectionState[Offset, Envelope](
         .futureSource(
           handlerStrategy.lifecycle
             .tryStart()
-            .flatMap(_ => sourceProvider.source(() => readOffsets())))
+            .flatMap { _ =>
+              sourceProvider
+                .source(() => readOffsets())
+                .map(_.mapMaterializedValue(_ => NotUsed))
```

Comment:
We need to map it to `NotUsed` over and over again. Wondering if we should instead change the signature to `Source[Envolope, NotUsed]`, it's now `Source[Envolope, _]`

Reply:
sounds good to change, since we don't use it

Reply:
Ok, I will add to this PR. It will impact all the current PRs, but better to have all there at once.

Reply:
So, there is two things here: 

1. the type returned by the SourceProvider which is still `Source[Done, _]` and we need to map it to something because of Scala 2.12. 
1. the type returned by `mappedSource` is now `Future[Done]`. This one we want it stable because is internal stuff and it will alway be `Future[Done]`.

I will do another iteration for change the type of the SourceProvider to `Source[Done, NotUsed]`,